packages/cinterop/src/darwin/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -2123,6 +2127,18 @@ actual object RealmInterop {
         return AuthProvider.of(realm_wrapper.realm_auth_credentials_get_provider(credentials.cptr()))
     }
 
+    actual fun realm_user_get_access_token(user: RealmUserPointer): String {
+        return realm_wrapper.realm_user_get_access_token(user.cptr())
```

Comment:
I think you need to use `safeKString` here and for the other strings, otherwise the underlying data might go out of scope.